### ISSUE TITLE
Database service is unhealthy after restart

### ISSUE THREAD

**Issue by martin-klima** [lando/lando#1381]
The database service starts only once just after rebuild and then database is unhealthy for all next starts in drupal7 recipe.

**Tell us about your setup**

Linux Mint 18.2
Linux 4.15.0-33-generic 36~16.04.1-Ubuntu SMP Wed Aug 15 17:21:05 UTC 2018 x86_64 GNU/Linux

**Lando v3.0.0-rc.2**

`docker version`
Client:
 Version:           18.09.0
 API version:       1.39

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
 
**Tell us about your `.lando.yml`**

Very simple test site.
```
name: my-lando-app
recipe: drupal7
config:
  webroot: .
```

**Tell us about the command you were running**

first run `lando start` - all is OK.
`lando restart` - see error below

**Tell us about the error you got**
```
info: Waiting until database service is ready...
warn: Service database is **unhealthy**
```
`lando mysql`
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (111)

**Tell us more**

The problem is probably related to drupal7 recipe. The drupal8 recipe works normally. I tested `lando poweroff, lando stop` without effect. When I use `lando rebuild`, the database container starts normally, untill `lando restart`.

**Comment by martin-klima** [comment]
This is part of log, maybe it helps.
```
lando logs -s database | grep database

database_1   | Version: '5.7.25'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL)
database_1   | mkdir: cannot create directory '/etc/ssh': Permission denied
database_1   | Running command /entrypoint.sh /run.sh
database_1   | 
database_1   | Welcome to the Bitnami mysql container
database_1   | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
database_1   | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
database_1   | Send us your feedback at containers@bitnami.com
database_1   | 
database_1   | INFO  ==> ** Starting MySQL setup **
database_1   | INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars..
database_1   | WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
database_1   | INFO  ==> Initializing mysql database...
database_1   | INFO  ==> Persisted data detected. Restoring...
database_1   | INFO  ==> Stopping mysql...
database_1   | Terminated
database_1   | mkdir: cannot create directory '/etc/ssh': Permission denied
database_1   | Running command /entrypoint.sh /run.sh
```

**Comment by pirog** [comment]
Im pushing a fix for this to master. Thanks for reporting @martin-klima

**Comment by martin-klima** [comment]
@pirog I gladly helped.
I extended my tests and I can confirm the problem also appears in Drupal 8 recipe, so it is not related to only D7 recipe as I supposed, but more probably is related to database service (image or config).

The problem is dynamic and not all restarts failed. I tested these steps in sequence for simple D8 recipe:
```
lando destroy -y
lando restart -> all OK
lando restart -> all OK
lando restart -> all OK
lando restart -> warn: Service database is unhealthy
```

**Comment by rabauss** [comment]
can confirm - that the problem randomly exists in RC3!

I use LAMP recipe with custom config for ``sql_mode``:
```
services:
    database:
        config:
            database: .lando/mysql/my.cnf
```
Here is the ErrorLog:

![image](https://user-images.githubusercontent.com/14016098/52171388-91e9bd80-275c-11e9-9e8e-faa2ffd44e6e.png)

**Comment by martin-klima** [comment]
I continued with tests on another PC with Linux Mint and the pattern is unpredictible. 1 = OK, 0 = Fail. The 1st run is after `lando destroy`.
```
Test1: 1101100110100110011101110 (10/25 fail)
Test2: 1000110000000000000001100 (20/25 fail)
Test3: 1000000000110000110000000(20/25 fail)
Test4: 1011000011101000000110000 (16/25 fail)
```
The next message in log
```
database_1   | mkdir: cannot create directory '/etc/ssh': Permission denied
database_1   | Running command /entrypoint.sh /run.sh
```
is always occurs regardless of the result. The only difference in log is:
```
database_1   | INFO  ==> ** Starting MySQL setup **
database_1   | INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars..
database_1   | WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
database_1   | INFO  ==> Initializing mysql database...
database_1   | INFO  ==> Persisted data detected. Restoring...
database_1   | INFO  ==> Stopping mysql...
database_1   | INFO  ==> ** MySQL setup finished! **
database_1   | 
database_1   | INFO  ==> ** Starting MySQL **
```
for started database and 
```
database_1   | INFO  ==> ** Starting MySQL setup **
database_1   | INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars..
database_1   | WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
database_1   | INFO  ==> Initializing mysql database...
database_1   | INFO  ==> Persisted data detected. Restoring...
database_1   | INFO  ==> Stopping mysql...
database_1   | Terminated
```
for failed database.

There are also 2 opened issues look very similar to our problem on bitnami-docker-mysql:
- https://github.com/bitnami/bitnami-docker-mysql/issues/47
- https://github.com/bitnami/bitnami-docker-mysql/issues/45

**Comment by martin-klima** [comment]
I renamed the issue title because the same problem is also in custom apps with services
`type: mysql`
`type: mysql:8.0`

e.g.:
```
name: testdbissue

services:
  appserver:
    type: mysql:8.0
```

and not related to Drupal recipes.

**Comment by pirog** [comment]
Ok yeah now i am definitely seeing this. Investigating.

**Comment by pirog** [comment]
Looks like this also happens for the `mariadb` service but not `postgres`.

**Comment by pirog** [comment]
Alright, pretty sure we've got this locked down. Needed to delegate with a new process from the lando entrypoint to the db entrypoint. whew ;)

**Comment by pirog** [comment]
Alright, ive got this in pending tests. We will likely roll a bug fix release later tonight before the wave hits Monday morning! Thanks to everyone for the feedback here

**Comment by pirog** [comment]
OK this is in master and havent run into it in about 500 trials. I THINK WE ARE BACK IN BIZNIZZ!

**Comment by martin-klima** [comment]
@pirok I can confirm the problem was fixed. Thanks. It had to be a busy weekend.

**Comment by sarath49** [comment]
I am facing this issue for postgres with rc5 release on again starting after shutdown on windows.

**Comment by akanelson** [comment]
I'm facing this issue with MySQL 5.7 using Lando v3.0.0-rc.12 on Mac

**Comment by No-Ssweat** [comment]
After updating to Lando v3.0.0-rc.13 on Windows, I am seeing this issue. I think I was on RC3 or RC4 before.

**Comment by shabananavas** [comment]
I'm getting this as well on the most basic D8 recipe.

**Comment by keithnoseworthy** [comment]
I'm also having this problem with v3.0.0-rc.13 running a D7 site locally on windows 10. I have not noticed it on D8. Running `destroy` and `start` seems to fix it.

**Comment by pirog** [comment]
Worth pointing out... AGAIN... that destroying and recreating your app _may_ be a necessary step and is bullet point numero uno in the update guide
https://docs.devwithlando.io/guides/updating-to-rc2.html#new-images

**Comment by keithnoseworthy** [comment]
> Worth pointing out... AGAIN... that destroying and recreating your app _may_ be a necessary step and is bullet point numero uno in the update guide
> https://docs.devwithlando.io/guides/updating-to-rc2.html#new-images

Sorry, above the update guide it says in bold above that point that "This guide is for people upgrading from 3.0.0-rc.1 or lower."  Is that a typo?

Thanks for you help on this.

**Comment by hkirsman** [comment]
For me it seemed also to change to mariadb instead of default mysql. I've also seen mariadb perform better but haven't tested with latest rc14. I also had to destroy the the app like said above. This is my db config now:

    database:
      type: mariadb:10.2

**Comment by micc83** [comment]
v3.0.0-rc.18

Same issue with Laravel recipe.

**Comment by habhatti** [comment]
Can confirm @micc83's report above. Additional details below:

The database service starts only once on a new project and then database is unhealthy for all next starts in the laravel recipe.

**Tell us about your setup**
Windows 10 Pro (x64)
Lando v3.0.0-rc.18

docker version
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:31 2019
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     false

**Tell us about your .lando.yml**
name: exampleproject
recipe: laravel
config:
  webroot: app/public


**Tell us about the command you were running**
first run lando start - all is OK.
lando restart - see error below

**Tell us about the error you got**
info: Waiting until database service is ready...
warn: Service database is **unhealthy**
lando mysql
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (111)

**Comment by pirog** [comment]
@habhatti @micc83 i'd recommend opening up a separate ticket because its unlikely this is the same thing as what @martin-klima originally posted

**Comment by habhatti** [comment]
@pirog #1719

**Comment by hlslaughter** [comment]
I'm seeing this (on lando v3.0.0-rc.20 / Docker version 19.03.1, build 74b1e89)  immediately after upgrading lando/docker.

I initially saw a wave of different errors that suggested my pre-existing containers were incompatible or something. After running 'docker prune' those errors cleared up and I saw the error mentioned here.

After doing a destroy/rebuild, dock, lando hanged on waiting for appserver.... but that could be related to a change in the configuration.

**Comment by marsjaninzmarsa** [comment]
Same happened to me with clean `wordpress` recipe on same versions as above, `mkdir: cannot create directory '/etc/ssh': Permission denied` followed by:
```
database_1   | 2019-08-29T08:31:45.254615Z 0 [Warning] Failed to set up SSL because of the following SSL library error: SSL context is not usable without certificate and private key
database_1   | 2019-08-29T08:31:45.254661Z 0 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306                                                                                                                                                                       database_1   | 2019-08-29T08:31:45.256066Z 0 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
database_1   | 2019-08-29T08:31:45.256350Z 0 [Note] Server socket created on IP: '0.0.0.0'.
database_1   | 2019-08-29T08:31:45.256989Z 0 [ERROR] Another process with pid 1 is using unix socket file.
database_1   | 2019-08-29T08:31:45.257029Z 0 [ERROR] Unable to setup unix socket lock file.
database_1   | 2019-08-29T08:31:45.257036Z 0 [ERROR] Aborting
```

**Comment by LPIX-11** [comment]
I think I found a solution
https://github.com/lando/lando/issues/1718

**Comment by yuseferi** [comment]
base on @LPIX-11 thread,
the following command works for me
`lando rebuild -s database`

**Comment by edutrul** [comment]
`lando rebuild -s database`
Worked for me

**Comment by faiwiz** [comment]
I faced same issue, issue was fixed just I upgrade docker and lando to latest version issue

**Comment by vesko-vujovic** [comment]
I have the same issue with lando v3.0.0.1 and docker 19.03. Any ideas? 

This is the message that I'm getting.

**_mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")'
Check that mysqld is running and that the socket: '/var/run/mysqld/mysqld.sock' exists!
ERROR ==> Check that mysqld is running and that the socket: '/var/run/mysqld/mysqld.sock' exists!_**

**Comment by yuseferi** [comment]
update your lando, right now the latest version is 3.0.4 and after that try
to `lando logs` to find out what is the problem.

>
